packages/front-end/components/Features/FeatureModal/index.tsx

```diff
@@ -113,13 +115,31 @@ const genFormDefaultValues = ({
   | "project"
   | "id"
   | "environmentSettings"
+  | "customFields"
 > => {
   const environmentSettings = genEnvironmentSettings({
     environments,
     featureToDuplicate,
     permissions: permissionsUtil,
     project,
   });
+  let customFieldValues = {};
+  if (customFields && featureToDuplicate) {
+    customFieldValues = featureToDuplicate.customFields
+      ? Object.fromEntries(
+          Object.entries(featureToDuplicate.customFields).filter(([key]) =>
+            customFields.some((a) => a.id.toString() === key)
+          )
+        )
+      : Object.fromEntries(
+          customFields.map((field) => [field.id.toString(), field.defaultValue])
+        ); // Create a fallback object
+  } else if (customFields) {
+    customFieldValues = Object.fromEntries(
+      customFields.map((field) => [field.id.toString(), field.defaultValue])
+    );
+  }
```

Comment:
This has some weird edge cases. For example, if the feature you are duplicating is missing some custom fields, this will not add them with default values.

You can fix that and simplify this code a lot by getting rid of the if/else and always looping through `customFields` and setting the value to `featureToDuplicate?.customFields?.[key] ?? field.defaultValue`

Also, what data type is the id that you're having to call to string on it? Seems very unusual.

Reply:
Ya, that would be simpler. I didn't want to persist a custom field value that was set but not shown as the filed ID does not apply to your project any more. The to string is not needed

Reply:
I don't quite follow.  `customFields` is already filtered to only include ones that are applicable to this feature.  So if you loop over that, you should be good.  This is what I was thinking:

```ts
const customFieldValues = customFields ? 
  Object.fromEntries(customFields.map((field) =>(
     [ field.id, featureToDuplicate?.customFields?.[key] ?? field.defaultValue ]
  )))
  : {};
```

Reply:
ya, that's cleaner. I'll make the change